Dev

FantasmasDev opened the pull request:
домашнее задание на проверку

FantasmasDev committed:
1. Добавлены кнопки и TextView

FantasmasDev committed:
1. Добавлены 2 экрана

FantasmasDev committed:
1. созданы экраны: share, libray
2. убрана заглушка toast
3. реализован переход между экранами

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsAcivity.kt:
```diff
@@ -0,0 +1,21 @@
+        val buttonGoHome = findViewById<ImageButton>(R.id.setting_home_button)
+        buttonGoHome.setOnClickListener {
+            val libraryIntent = Intent(this, MainActivity::class.java)
+            startActivity(libraryIntent)
```
Для перехода назад надо использовать не интент, а методы finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на экран настроек. На эту тему можно почитать про стек активити. Это будет критично в следующем спринте

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,83 @@
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+    <FrameLayout
```
Лишняя вложенность, FrameLayout тут не нужен

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,83 @@
+            android:layout_gravity="start"
+            android:text="@string/main_header"
+            android:textColor="@color/white"
+            android:textSize="22sp" />
```
Все повторяющиеся аналогичные размеры должны быть вынесены в ресурсы

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,83 @@
+            android:layout_height="0dp"
+            android:layout_marginBottom="8dp"
+            android:layout_weight="1"
+            android:backgroundTint="#fff"
```
Все цвета должны быть вынесены в ресурсы

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,128 @@
+        android:orientation="horizontal"
+        android:layout_marginTop="12dp"
+        android:paddingStart="16dp"
+        android:paddingTop="21dp"
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,128 @@
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:layout_marginTop="12dp"
+        android:paddingStart="16dp"
```
Лучше не прописывать отступы в каждом элементе, а задать общий отступ в родительском лейауте

faritowich commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,14 @@
+    <string name="dark_theme_switcher">Тёмная тема</string>
+    <string name="share_app">Поделиться приложением</string>
+    <string name="support_button">Написать в поддержку</string>
+    <string name="terms_of_use">Пользовательское соглашение</string>
```
Сейчас в проекте два файла со строками на русском. Надо один убрать, один оставить

FantasmasDev committed:
переработаны файлы ресурсов
1. добавлены размеры.
2. изменён метод возврата на основной экран
3. доработана подстановка цвета из ресурсов
4. подстановка размеров из ресурсов
